packages/protocol/src/client-server-protocol/jsonrpc/base-jsonrpc-glsp-client.ts

```diff
@@ -40,49 +35,41 @@ export class BaseJsonrpcGLSPClient implements GLSPClient {
     }
 
     shutdownServer(): void {
-        if (this.checkConnectionState()) {
-            this.resolvedConnection!.sendNotification(JsonrpcGLSPClient.ShutdownNotification);
-        }
+        this.checkedConnection.sendNotification(JsonrpcGLSPClient.ShutdownNotification);
     }
 
     initializeServer(params: InitializeParameters): Promise<InitializeResult> {
-        if (this.checkConnectionState()) {
-            return this.resolvedConnection!.sendRequest(JsonrpcGLSPClient.InitializeRequest, params);
-        }
-        return Promise.reject(JsonrpcGLSPClient.ClientNotReadyMsg);
+        return this.checkedConnection.sendRequest(JsonrpcGLSPClient.InitializeRequest, params);
     }
 
     initializeClientSession(params: InitializeClientSessionParameters): Promise<void> {
-        if (this.checkConnectionState()) {
-            return this.resolvedConnection!.sendRequest(JsonrpcGLSPClient.InitializeClientSessionRequest, params);
-        }
-        return Promise.reject(JsonrpcGLSPClient.ClientNotReadyMsg);
+        return this.checkedConnection.sendRequest(JsonrpcGLSPClient.InitializeClientSessionRequest, params);
     }
 
     disposeClientSession(params: DisposeClientSessionParameters): Promise<void> {
-        if (this.checkConnectionState()) {
-            return this.resolvedConnection!.sendRequest(JsonrpcGLSPClient.DisposeClientSessionRequest, params);
-        }
-        return Promise.reject(JsonrpcGLSPClient.ClientNotReadyMsg);
+        return this.checkedConnection.sendRequest(JsonrpcGLSPClient.DisposeClientSessionRequest, params);
     }
 
-    onActionMessage(handler: ActionMessageHandler): void {
-        if (this.checkConnectionState()) {
-            this.resolvedConnection!.onNotification(JsonrpcGLSPClient.ActionMessageNotification, handler);
-        }
+    onActionMessage(handler: ActionMessageHandler): Disposable {
+        return this.checkedConnection.onNotification(JsonrpcGLSPClient.ActionMessageNotification, handler);
     }
 
     sendActionMessage(message: ActionMessage): void {
-        if (this.checkConnectionState()) {
-            this.resolvedConnection!.sendNotification(JsonrpcGLSPClient.ActionMessageNotification, message);
-        }
+        this.checkedConnection.sendNotification(JsonrpcGLSPClient.ActionMessageNotification, message);
     }
 
-    protected checkConnectionState(): boolean {
+    protected get checkedConnection(): MessageConnection {
         if (!this.isConnectionActive()) {
             throw new Error(JsonrpcGLSPClient.ClientNotReadyMsg);
         }
-        return true;
+        return this.resolvedConnection!;
+    }
+
+    /**
+     * @Deprecated use {@link checkConnection} instead
+     */
+    protected checkConnectionState(): boolean {
+        return this.checkedConnection !== undefined;
```

Comment:
Since this is only for backwards compatibility I think having this method is fine. In the long run it would be great to have one method that throws an exception (checkedConnection) and one that does a safe check, i.e., always return a boolean. A mixture of boolean and exception is unexpected.

Reply:
Yeah, we already have this two methods with
`checkedConnection` and `isConnectionActive`. 
As you already mentioned the `checkConnectionState()` method is just for backwards compatibility. But I would also fine with dropping this method entirely.

Reply:
Since you marked it as deprecated, it is fine for me. I just wanted to note down my thoughts because it is indeed an unexpected method. Thank you for the clarification.